core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1565,6 +1647,39 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
         ErrorReport(StatusCodes.BadRequest, "may not grant readers compute access")
       )
     }
+    if (
+      workspace.workspaceType.equals(WorkspaceType.McWorkspace) &&
+      aclChanges.exists {
+        case WorkspaceACLUpdate(_, WorkspaceAccessLevels.Write, _, Some(true)) => true
+        case _                                                                 => false
+      }
+    ) {
+      throw new RawlsExceptionWithErrorReport(
```

Comment:
Probably out of scope for this PR but food for thought: I've been trying to move toward throwing more targeted exception types rather than the catch-all RawlsExceptionWithErrorReport. It makes the code and associated tests more readable and I think it prevents us from missing edge cases in the tests (i.e., masking a real bug if a lower level piece of code throws `RawlsExceptionWithErrorReport`)